File: sentimental-credit/credit.py
# TODO
from cs50 import get_int


h = get_int("Number: ")

sum = 0
counter = 0

while (h > 0):
    counter += 1
    if (h < 99 and h > 9):
        dd = h
    d = h % 10
    if (counter % 2 != 0):
        sum = sum + d
    else:
        double_d = d * 2
        if (double_d > 9):
            double_d = double_d % 10
            double_d += 1
        sum = sum + double_d
    h = int(h / 10)

if (sum % 10 != 0):
    print("INVALID\n")
else:
    if (d == 4 and (counter == 13 or counter == 16)):
        print("VISA\n")
    elif (d == 3 and (dd == 34 or dd == 37) and counter == 15):
        print("AMEX\n")
    elif (d == 5 and counter == 16):
        if (dd == 51 or dd == 52 or dd == 53 or dd == 54 or dd == 55):
            print("MASTERCARD\n")
        else:
            print("INVALID\n")
    else:
        print("INVALID\n")

File: credit/credit.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    long h = get_long("Number: ");

    //total sum
    int sum = 0;
    // counter to know if you are on odd or even digit
    int counter = 0;
    // current last digit
    int d;
    // used to find first 2 digit to determine type of card
    int dd;

    while (h > 0)
    {
        counter ++;
        if (h < 99 && h > 9)
        {
            dd = h;
        }
        d = h % 10;
        if (counter % 2 != 0)
        {
            sum = sum + d;
        }
        else
        {
            int double_d = d * 2;
            if (double_d > 9)
            {
                double_d = double_d % 10;
                double_d ++;
            }
            sum = sum + double_d;
        }
        h = h / 10;
    }

    if (sum % 10 != 0)
    {
        printf("INVALID\n");
    }
    else
    {
        if (d == 4 && (counter == 13 || counter == 16))
        {
            printf("VISA\n");
        }
        else if (d == 3 && (dd == 34 || dd == 37) && counter == 15)
        {
            printf("AMEX\n");
        }
        else if (d == 5 && counter == 16)
        {
            if (dd == 51 || dd == 52 || dd == 53 || dd == 54 || dd == 55)
            {
                printf("MASTERCARD\n");
            }
            else
            {
                printf("INVALID\n");
            }
        }
        else
        {
            printf("INVALID\n");
        }
    }
}